### PR TITLE
Introduce validation that checks for duplicate package names

### PR DESCRIPTION
**Pull request by adamkpickering**
Packages (i.e. packages/vendor/name) are namespaced by vendor, but the corresponding charts that are eventually served from the partner charts repository are not. The charts' names come from the package names. So, we need to ensure that users cannot create a package with a name that another package already has. This PR adds this check.

In order to make this PR work, I also had to move `PackageWrapper` to its own package. I also had to move the existing validation code into its own function - the idea here is to create a validation function for each thing we want to check about the partner charts repository, and then call those functions in a list.

Again, reviewing by commit will make the most sense here.

### CODE REVIEW

File: pkg/upstreamyaml/upstreamyaml.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func Parse(upstreamYamlPath string) (*UpstreamYaml, error) {
 	return upstreamYaml, nil
 }
 
-func Write(upstreamYamlPath string, upstreamYaml UpstreamYaml) error {
+func Write(upstreamYamlPath string, upstreamYaml *UpstreamYaml) error {
```
Comment (diogoasouza) [inline pkg/upstreamyaml/upstreamyaml.go:121]: Is this change intended?

Comment (adamkpickering) [reply pkg/upstreamyaml/upstreamyaml.go:121]: Yes, it is intended. It isn't a functional change - the only purpose is to avoid having to use the `*` operator to dereference `packageWrapper.UpstreamYaml` where `upstreamyaml.Write()` is called in main.go. `packageWrapper.UpstreamYaml` is of type `*upstreamyaml.UpstreamYaml`.

**Review by diogoasouza — APPROVED** [review]
